keps/sig-network/4962-network-topology-standard/README.md

```diff
@@ -0,0 +1,938 @@
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This document proposes a standard for declaring cluster network topology in Kubernetes, representing the hierarchy of nodes, switches, and interconnects. In this context, a switch can refer to a physical network device or a collection of such devices with close proximity and functionality.
+
+## Motivation
+
+Understanding the cluster network topology is essential for optimizing the placement of workloads that require intensive inter-node communication. Currently, there is no standardized way to represent this information in Kubernetes, making it challenging to develop control plane components and applications that can leverage topology awareness.
+
+This information might be useful for various components and features, including:
+
+- Pod affinity sections in deployment and pod specs
+- Kueue topology-aware scheduling
+- Development of native scheduler plugins for topology-aware scheduling, for example:
+  - Topology-aware gang-scheduling scheduler plugin
+  - Gang-scheduling auto-scaler
+  - DRA scheduler plugin
+
+### Goals
+
+- Introduce a standard way of representing network topology in Kubernetes clusters
+
+### Non-Goals
+
+- Implement a topology-aware gang-scheduling scheduler plugin
+- Define or implement topology discovery mechanisms for CSPs or on-premises environments
+
+## Proposal
+
+We propose new node label and annotation types to capture network topology information:
+- Network Topology Label
+- Network QoS Annotation
+
+### Network Topology Label
+Format: `network.topology.kubernetes.io/<nw-switch-type>: <switch-name>`
+- `<nw-switch-type>`: Logical type of the network switch (can be one of the reserved names or a custom name)
+  - Reserved names: `accelerator`, `block`, `datacenter`, `zone`
+- `<switch-name>`: Unique identifier for the switch
+
+### Network QoS Annotation
+Format: `network.qos.kubernetes.io/switches: <QoS>`
```

Comment:
This is not really the switch, right, is the interface in the node that connects to the switch ... and we already have properties to define attributes on the network interfaces with DRA, see slice 14 https://docs.google.com/presentation/d/1Vdr7BhbYXeWjwmLjGmqnUkvJr_eOUdU0x-JxfXWxUT8/edit#slide=id.g2f750386db2_5_0 so I don't feel we need this additional annotation here